### PR TITLE
Merge from ILM master

### PR DESCRIPTION
**Pull request by bernardkwok**
- switch to row major ordering for matrices
- updates for lighting/shadowing/baking
- STB image library update from 2.19 to 2.23.
- image resources folder changes
- cleanup in GLSL support code including GLFrameBuffer class extraction

Test results:
[tests.pdf](https://github.com/autodesk-forks/MaterialX/files/4294519/tests.pdf)

### CODE REVIEW

File: resources/Materials/TestSuite/_options.mtlx [hunk 1]
```diff
@@ -78,20 +78,20 @@
...
-      <parameter name="radianceIBLPath" type="string" value="resources/Images/Environments/san_giuseppe_bridge.hdr" />
+      <parameter name="radianceIBLPath" type="string" value="resources/Lights/san_giuseppe_bridge.hdr" />
 
       <!-- Suggested irradiance IBL file path -->
-      <parameter name="irradianceIBLPath" type="string" value="resources/Images/Environments/san_giuseppe_bridge_diffuse.hdr" />
+      <parameter name="irradianceIBLPath" type="string" value="resources/Lights/irradiance/san_giuseppe_bridge.hdr" />
 
       <!-- Transforms UVs of loaded geometry -->
-      <parameter name="transformUVs" type="matrix44" value="1.0f, 0.0f, 0.0f, -0.235f, 0.0f, -1.0f, 0.0f, 1.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f" />
+      <parameter name="transformUVs" type="matrix44" value="1.0f, 0.0f, 0.0f, 0.0f, 0.0f, -1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, -0.235f, 1.0f, 0.0f, 1.0f" />
```
Comment (bernardkwok) [inline resources/Materials/TestSuite/_options.mtlx]: This is the extra uv transform we did to align with OSL geometry. It's column order and now the matrix library is row order so it's transposed.

File: resources/Materials/TestSuite/lights/envmap_shader.mtlx [hunk 2]
```diff
@@ -13,7 +13,7 @@
          <input name="in2" type="float" nodename="envUvs" channels="y"/>
       </combine2>
       <image name="envImage" type="color3">
-         <parameter name="file" type="filename" value="resources/Images/Environments/san_giuseppe_bridge.hdr" />
+         <parameter name="file" type="filename" value="resources/Lights/san_giuseppe_bridge.hdr" />
```
Comment (bernardkwok) [inline resources/Materials/TestSuite/lights/envmap_shader.mtlx]: There's a bunch of these as IBL file locations changed.

File: resources/Materials/TestSuite/stdlib/color_management/color3_vec3_cm_test.mtlx [hunk 3]
```diff
@@ -1,40 +1,40 @@
...
-            <parameter name="index" type="integer" value="1"/>
-        </extract>
-        <heighttonormal name="impl_heighttonormalmap" type="vector3">
-            <input name="in" type="float" nodename="extract1"/>
-            <parameter name="scale" type="float" value="1.0"/>
-        </heighttonormal>
-        <output name="height_normal_map_output" type="vector3" nodename="impl_heighttonormalmap"/>
-    </nodegraph>
+  <!-- create nodegraph to test cm conversion on image node before feeding to heighttonormal  -->
+  <nodegraph name="height_to_norma_cm">
```
Comment (feldstj) [inline resources/Materials/TestSuite/stdlib/color_management/color3_vec3_cm_test.mtlx]: minor: probably not worth mentioning as it was wrong before, but should it be height_to_normal_cm?

File: resources/Materials/TestSuite/stdlib/color_management/filename_cm_test.mtlx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0"?>
 <materialx version="1.37" colorspace="lin_rec709">
+
```
Comment (feldstj) [inline resources/Materials/TestSuite/stdlib/color_management/filename_cm_test.mtlx]: minor: Should this new line be removed?

**Pull request merged**